src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -158,8 +158,10 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
                 interfaceDocumentId,
                 cancellationToken);
 
-            var solutionWithFormattedInterfaceDocument = GetSolutionWithFormattedInterfaceDocument(unformattedInterfaceDocument, cancellationToken);
-
+            var syntaxFactsService = refactoringResult.DocumentToExtractFrom.GetLanguageService<ISyntaxFactsService>();
```

Comment:
I think you need to change
http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/ExtractInterface/AbstractExtractInterfaceService.cs,58
to Async

and change all methods below as Async. and then change caller of the ExtrctInterface(Async) to use WaitAndGetResult.